code/modules/power/lighting.dm

```diff
@@ -575,6 +575,8 @@ var/global/list/obj/machinery/light/alllights = list()
 /obj/machinery/light/power_change()
 	spawn(10)
 		var/area/this_area = get_area(src)
+		if(this_area == null)
```

Comment:
Same with the APC, this feels like the light is being nullspaced or otherwise put in the garbage collector, but a reference is being held somewhere that prevents it from being deleted.

Find what could be holding references to /obj/machinery/light